### ISSUE TITLE
db.LastInsertRowID() produces strange results.

### ISSUE THREAD

**Issue by kuroneko** [kuroneko/gosqlite3#5]
From github message:

Chris,

I am having a problem with gosqlite3. I am unsure if it is a bug or not so I didn't want to create an issue. Here is C code:

```
#include 
#include

int main(int argc, char **argv) {
sqlite3 *db;

unlink("test.db3");

sqlite3_open("test.db3", &db);
sqlite3_exec(db, "CREATE TABLE people (id INTEGER PRIMARY KEY AUTOINCREMENT, name TEXT)", NULL, NULL, NULL);   
sqlite3_exec(db, "INSERT INTO people (name) VALUES ('John')", NULL, NULL, NULL);
printf("id=%d\n", sqlite3_last_insert_rowid(db));

sqlite3_close(db);
}
```

The output is "id=1"

Here is Go code:

```
package main

import (
  "os"
  "fmt"

  db "github.com/kuroneko/sqlite3"
)

func main() {
  os.Remove("test.db3")

  d, _ := db.Open("test.db3")
  d.Execute("CREATE TABLE people (id INTEGER PRIMARY KEY AUTOINCREMENT, name TEXT)")
  d.Execute("INSERT INTO people (name) VALUES ('Jim Doe')")
  fmt.Printf("id=%v\n", d.LastInsertRowID())

  d.Close()
}
```

The output is "id=4294967299"

**Comment by kuroneko** [comment]
Works for me. :(

**Comment by jcowgar** [comment]
Since I initially contacted you I've done more research on this bug and it's not with your wrapper. It occurs on other Sqlite wrappers as well but only in Windows. On Linux it works great. I have put together a very small test case that wraps Open, Close, Exec and LastInsertRowID along with the Go and C code I provided to you. I sent an email to the go mailing list hoping someone would have an answer as to why this is happening. The C code works great the Go code fails. You can view that code at: http://github.com/jcowgar/sqlite3gotest ...

So, to recap. Everything works fine in Linux. GetLastRowID fails in Windows but not only with your SQLite3 wrapper but other SQLite3 wrappers. The C example code works in fine in Windows. Thus, I think it's pointing to a Windows Go bug. You can do as you wish with this ticket. It's not a direct problem of your code _however_ it does affect anyone using your library on Windows. So, your call.

**Comment by kuroneko** [comment]
Right - that explains a lot.  Windows Go is still under development and it's not surprising that cgo is not working correctly at this time.  You may wish to report this to them.

A list of known windows go issues is at : http://code.google.com/p/go/issues/list?q=label:os-windows
You should be able to submit bug reports there too.

**Comment by jcowgar** [comment]
The Go developers figured out the problem, it has to do with structure packing. They are going to make a fix but for now adding -fpack-struct to CGO_CFLAGS in the Makefile makes things work properly on Windows. Just a FYI in case any others report this problem before it is fixed in Go.
